source/common/router/config_impl.cc

```diff
@@ -600,6 +610,28 @@ void RouteEntryImplBase::finalizeResponseHeaders(Http::ResponseHeaderMap& header
   }
 }
 
+Http::HeaderTransforms
+RouteEntryImplBase::responseHeaderTransforms(const StreamInfo::StreamInfo& stream_info) const {
+  Http::HeaderTransforms transforms;
+  if (!vhost_.globalRouteConfig().mostSpecificHeaderMutationsWins()) {
+    // Append user-specified request headers from most to least specific: route-level headers,
+    // virtual host level headers and finally global connection manager level headers.
+    mergeTransforms(transforms, response_headers_parser_->getHeaderTransforms(stream_info));
+    mergeTransforms(transforms, vhost_.responseHeaderParser().getHeaderTransforms(stream_info));
+    mergeTransforms(
+        transforms,
+        vhost_.globalRouteConfig().responseHeaderParser().getHeaderTransforms(stream_info));
+  } else {
+    // Most specific mutations take precedence.
```

Comment:
Could you expand this comment with examples? Thanks!

Reply:
I copied this from `RouteEntryImplBase::finalizeResponseHeaders` so don't have any good examples, but I expanded the comment a bit.